linkbox: fix work with names longer than 8-25 Unicode chars.

gvitali opened the pull request:
#### What is the purpose of this change?

LinkBox API does not allow searching by more than 25 Unicode characters in the name, for this reason it is currently impossible to work with files and folders named longer than 8 Unicode chars (if encoded in base32). This fix queries all files in a long named directory and checks names there one by one, thus solving the issue.

#### Was the change discussed in an issue or in the forum before?

#7542

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

ncw commented on backend/linkbox/linkbox.go:
```diff
@@ -238,6 +239,14 @@ func (f *Fs) listAll(ctx context.Context, dirID string, name string, fn listAllF
 		name = ""
 	}
+
+	nameSearch := name
```
Instead of doing this you can just limit the maximum length that is OK in the `searchOK` regexp above

Something like

```go
var searchOK = regexp.MustCompile(`^[a-zA-Z0-9_ .]{1,50}$`)
```

gvitali replied:
Thank you for your comment. Yes, you're right, it's more concise.

ncw commented on fstest/fstests/fstests.go:
```diff
@@ -701,11 +702,15 @@ func Run(t *testing.T, opt *Opt) {
 				{"trailing dot", "trailing dot."},
 				{"invalid UTF-8", "invalid utf-8\xfe"},
 				{"URL encoding", "test%46.txt"},
+				{"many chars", "quite-long-file-name-with-more-than-fifty-chars.txt"},
```
Does this make the tests pass?

I note these are failing in the integration tests at the moment. (I'm sure they were passing when we merged linkbox so something has changed!)

gvitali replied:
This caused an error for Linkbox before the fix, but worked after the fix. However, it is better to move this into a separate commit, as it affects the tests of other remotes.

ncw reviewed (commented):
Very useful - thank you :-)

See comments inline.

gvitali commented:
During Linkbox synchronization I encountered two more problems and fixed them. The maximum number of objects on a page should be no more than 1000. Currently it is 1024, for this reason the listing always ends on the first page with the error “object not found”, rclone tries to upload the file again, Linkbox stores it with the name “filename(N)”, and so the storage fills up indefinitely.
In the second case, I added a hyphen to the list of allowed characters, which makes queries more optimized (no need to load all files in a directory for an entity with a hyphen).

azowrle commented:
Why is this still not merged? I really need to use Telebox with this fix

ncw commented:
I rebased this and squashed it down to two commits.

I'll merge when the CI goes green - thank you @gvitali

ncw commented:
I rebased this again and fixed the conflicts. Hopefully I'll remember to merge it this time!
